lib/collection/group.ts

```diff
@@ -11,17 +9,29 @@ export type Index = Array<PrimaryKey>;
 export type InstanceContext = (() => Collection) | (() => Pulse);
 
 export class Group<DataType = DefaultDataItem> extends State<Array<PrimaryKey>> {
-  _masterOutput: Array<DataType> = [];
-  missingPrimaryKeys: Array<PrimaryKey> = [];
+	_output: Array<DataType> = [];
+	_states: Array<State<DataType>> = []; // States of the Group
```

Comment:
I don't want to store direct references to the State items inside a group, it's bloat and breaks the ability to cleanly stringify a Group object

Reply:
yeah.. The main reason why I created this 'states' change.. was to fix the collection state rerender issue.. but there we found a better solution.. so it was just an idea which I thought could be pushed